src/Test.EndToEnd.Baselining/Program.cs

```diff
@@ -0,0 +1,123 @@
+    ///     CloudMine-Spam\...            
+    ///       mseng.log                    [Change the code in BaseliningDetailEnricher to add the Result properties you need to investigate differences quickly]
+    ///   
+    ///   Expected_Debug\                  [Each run will create an 'enriched' version of the Expected logs also]
+    ///     
+    /// 
+    ///  Execution Steps
+    ///  ===============
+    ///  Running tests will find each series folder in Input and:
+    ///    - Load the initial baseline
+    ///    - Load each log in order
+    ///    - Replace the GUID with a 'RID' (LogIndex + ' ' + ResultIndex)
+    ///    - Baseline the current log to the Baseline-so-far
+    ///    - Write detail report lines about how each Result was baselined
+    ///    - Save the output as the Baseline-so-far
+    ///    - Compare the output to the same file in 'Expected\', if present
+    ///    - Write 'enriched' logs to Output_Debug\ and Expected_Debug\ for human investigation.
+    ///  
+    ///    
+    ///  Usage
+    ///  =====
+    ///   Get Test Content:
+    ///   git clone https://github.com/microsoft/sarif-sdk-test-content
+    ///   
+    ///   Run a Full Pass:
+    ///   "Test.EndToEnd.Baselining run C:\Code\sarif-sdk-test-content\BaselineE2E"
+    ///   
+    ///   Debug matching for Result 004 in log 001 in series "Spam\v1\WorkItem\mseng":
+    ///   To quickly debug a Result, copy and paste the first line from the Series log for the Series path and the RID (ex: 001 004) from the Result to debug.
+    ///   "Test.EndToEnd.Baselining debug C:\Code\sarif-sdk-test-content\BaselineE2E Spam\v1\WorkItem\mseng 001 004"
+    ///   
+    ///   Re-generate debug logs with the details as coded in BaseliningDetailEnricher:
+    ///   "Test.EndToEnd.Baselining rebuild-debug-logs C:\Code\sarif-sdk-test-content\BaselineE2E"
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            try
+            {
+                Program program = new Program();
+
+                return Parser.Default.ParseArguments<RunOptions, DebugOptions, RebuildDebugLogsOptions>(args).MapResult(
+                    (RunOptions options) => Program.Run(options),
+                    (DebugOptions options) => Program.Debug(options),
+                    (RebuildDebugLogsOptions options) => Program.CreateDebugLogs(options),
+                    errs => 1);
+            }
+            catch (Exception ex) when (!Debugger.IsAttached)
+            {
+                Console.WriteLine(ex.ToString());
+                return 2;
+            }
+        }
+
+        static int Run(RunOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            BaseliningSummary overallSummary = tester.RunAll(options.TestRootPath);
+            Console.WriteLine($"{overallSummary}");
```

Comment:
>$"{overallSummary}" [](start = 30, length = 19)

Nit: Isn't this the same as `Console.WriteLine(overallSummary)`?

Reply:
Yes. This is what evolves when BaseliningSummary.ToString didn't always exist. =) Fixed.